### ISSUE TITLE
Unable to create sample cluster as per the startup guide

### ISSUE THREAD

**Issue by babu-selector** [kubernetes-sigs/cluster-api-provider-aws#470]
/kind bug

**What steps did you take and what happened:**

sigs.k8s.io/cluster-api-provider-aws $ clusterctl create cluster -v2 --provider aws \
>   -m ./cmd/clusterctl/examples/aws/out/machines.yaml \
>   -c ./cmd/clusterctl/examples/aws/out/cluster.yaml \
>   -p ./cmd/clusterctl/examples/aws/out/provider-components.yaml \
>   -a ./cmd/clusterctl/examples/aws/out/addons.yaml
I0105 13:35:51.741185    9064 plugins.go:39] Registered cluster provisioner "aws"
I0105 13:35:51.745170    9064 createbootstrapcluster.go:28] Creating bootstrap cluster
I0105 13:37:51.373475    9064 minikube.go:62] Ran: minikube [start --bootstrapper=kubeadm] Output: Starting local Kubernetes v1.10.0 cluster...
Starting VM...
Getting VM IP address...
E0105 13:36:45.674421    9065 start.go:210] Error parsing version semver:  Version string empty
Moving files into cluster...
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Stopping extra container runtimes...
Starting cluster components...
Verifying kubelet health ...
Verifying apiserver health ...Kubectl is now configured to use the cluster.
Loading cached images from config file.

Everything looks great. Please enjoy minikube!
I0105 13:37:51.375705    9064 clusterdeployer.go:95] Applying Cluster API stack to bootstrap cluster
I0105 13:37:51.375730    9064 applyclusterapicomponents.go:27] Applying Cluster API Provider Components
I0105 13:37:51.375743    9064 clusterclient.go:521] Waiting for kubectl apply...
I0105 13:37:51.946531    9064 clusterclient.go:550] Waiting for Cluster v1alpha resources to become available...
I0105 13:37:51.956530    9064 clusterclient.go:563] Waiting for Cluster v1alpha resources to be listable...
I0105 13:37:51.969380    9064 clusterdeployer.go:100] Provisioning target cluster via bootstrap cluster
I0105 13:37:51.976881    9064 applycluster.go:37] Creating cluster object test1 in namespace "default"
I0105 13:37:51.982603    9064 clusterdeployer.go:109] Creating master controlplane-0 in namespace "default"
I0105 13:37:51.989567    9064 applymachines.go:37] Creating machines in namespace "default"
I0105 13:37:52.003391    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:38:02.006575    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:38:12.009821    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:38:22.008634    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:38:32.008872    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:38:42.012856    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:38:52.012704    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:39:02.014276    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:39:12.010374    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:39:22.011445    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 13:39:32.014558    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...

….
..
…

I0105 14:06:02.015884    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:06:12.016296    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:06:22.016155    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:06:32.012275    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:06:42.015136    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:06:52.012351    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:02.014802    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:12.019052    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:22.023176    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:32.022164    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:42.026721    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:52.023787    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:52.026375    9064 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0105 14:07:52.028444    9064 createbootstrapcluster.go:37] Cleaning up bootstrap cluster.
I0105 14:07:52.499120    9064 minikube.go:62] Ran: minikube [delete] Output: Deleting local Kubernetes cluster...
Machine deleted.
F0105 14:07:52.499272    9064 create_cluster.go:64] unable to create master machine: timed out waiting for the condition

**What did you expect to happen:**

**Anything else you would like to add:**

I0105 21:55:43.980810       1 controller.go:89] Running reconcile Cluster for test1
I0105 21:55:43.980850       1 controller.go:124] reconciling cluster object test1 triggers idempotent reconcile.
I0105 21:55:43.980856       1 actuator.go:54] Reconciling cluster test1
I0105 21:55:43.981477       1 network.go:25] Reconciling network
I0105 21:55:43.981573       1 vpc.go:39] Reconciling VPC
I0105 21:55:44.694775       1 vpc.go:54] Working on VPC "vpc-07d9c9b7914bb07cc"
I0105 21:55:44.694808       1 subnets.go:38] Reconciling subnets
I0105 21:55:45.095892       1 subnets.go:116] Subnets available: [id=subnet-01543e17a9d7fa56c/az=us-east-2a/public=false id=subnet-03f2b6e6ef498e4e9/az=us-east-2a/public=true]
I0105 21:55:45.095984       1 gateways.go:33] Reconciling internet gateways
I0105 21:55:45.349830       1 natgateways.go:34] Reconciling NAT gateways
I0105 21:55:45.593065       1 routetables.go:36] Reconciling routing tables
I0105 21:55:45.736291       1 routetables.go:45] Subnet "subnet-01543e17a9d7fa56c" is already associated with route table "rtb-0456c67a8c0fc02e6"
I0105 21:55:45.737033       1 routetables.go:45] Subnet "subnet-03f2b6e6ef498e4e9" is already associated with route table "rtb-09d6f14ea76122f8d"
I0105 21:55:45.737043       1 securitygroups.go:35] Reconciling security groups
I0105 21:55:45.994951       1 securitygroups.go:71] Security group for role "bastion": id=sg-075108744364f3201/name=test1-bastion
I0105 21:55:45.994996       1 securitygroups.go:71] Security group for role "controlplane": id=sg-000af8804c0718c1d/name=test1-controlplane
I0105 21:55:45.995007       1 securitygroups.go:71] Security group for role "node": id=sg-099f1b2e964bb698d/name=test1-node
I0105 21:55:45.995408       1 network.go:57] Reconcile network completed successfully
I0105 21:55:45.995420       1 bastion.go:41] Reconciling bastion host
I0105 21:55:46.293636       1 bastion.go:70] Reconcile bastion completed successfully
I0105 21:55:46.293654       1 loadbalancer.go:36] Reconciling load balancers
E0105 21:55:47.497381       1 scope.go:159] [scope] failed to store provider status for cluster "test1" in namespace "default": the server could not find the requested resource (put clusters.cluster.k8s.io test1)
E0105 21:55:47.497423       1 controller.go:131] Error reconciling cluster object test1; unable to reconcile load balancers: failed to create classic load balancer: &{test1-apiserver  Internet-facing [subnet-03f2b6e6ef498e4e9] [sg-000af8804c0718c1d] [0xc0009a6720] 0xc0009a6750 map[sigs.k8s.io/cluster-api-provider-aws/managed:true sigs.k8s.io/cluster-api-provider-aws/role:apiserver kubernetes.io/cluster/test1:owned]}: AccessDenied: User: arn:aws:iam::962279257951:user/bootstrapper.cluster-api-provider-aws.sigs.k8s.io is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::962279257951:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
	status code: 403, request id: a85d294e-1134-11e9-8b99-25ecf1fa4647

I0105 22:01:25.754852       1 controller.go:114] Running reconcile Machine for controlplane-0
I0105 22:01:25.755231       1 actuator.go:236] Checking if machine controlplane-0 for cluster test1 exists
E0105 22:01:25.798837       1 machine_scope.go:129] [machinescope] failed to store provider status for machine "controlplane-0" in namespace "default": the server could not find the requested resource (put machines.cluster.k8s.io controlplane-0)
I0105 22:01:25.798912       1 controller.go:176] Reconciling machine object controlplane-0 triggers idempotent create.
I0105 22:01:25.798935       1 actuator.go:64] Creating machine controlplane-0 for cluster test1
E0105 22:01:26.300194       1 machine_scope.go:129] [machinescope] failed to store provider status for machine "controlplane-0" in namespace "default": the server could not find the requested resource (put machines.cluster.k8s.io controlplane-0)
W0105 22:01:26.300224       1 controller.go:178] unable to create machine controlplane-0: failed to retrieve controlplane url during machine creation: no classic load balancer found with name: "test1-apiserver": LoadBalancerNotFound: There is no ACTIVE Load Balancer named 'test1-apiserver'
	status code: 400, request id: 726f5ac8-1135-11e9-9041-0910b09c2248
I0105 22:06:42.892877       1 controller.go:89] Running reconcile Cluster for test1
I0105 22:06:42.892903       1 controller.go:124] reconciling cluster object test1 triggers idempotent reconcile.
I0105 22:06:42.892909       1 actuator.go:54] Reconciling cluster test1
I0105 22:06:42.893666       1 network.go:25] Reconciling network
I0105 22:06:42.893698       1 vpc.go:39] Reconciling VPC
I0105 22:06:43.575439       1 vpc.go:54] Working on VPC "vpc-07d9c9b7914bb07cc"
I0105 22:06:43.575456       1 subnets.go:38] Reconciling subnets
I0105 22:06:43.874058       1 subnets.go:116] Subnets available: [id=subnet-01543e17a9d7fa56c/az=us-east-2a/public=false id=subnet-03f2b6e6ef498e4e9/az=us-east-2a/public=true]
I0105 22:06:43.874081       1 gateways.go:33] Reconciling internet gateways
I0105 22:06:44.016618       1 natgateways.go:34] Reconciling NAT gateways
I0105 22:06:44.256988       1 routetables.go:36] Reconciling routing tables
I0105 22:06:44.389887       1 routetables.go:45] Subnet "subnet-01543e17a9d7fa56c" is already associated with route table "rtb-0456c67a8c0fc02e6"
I0105 22:06:44.389910       1 routetables.go:45] Subnet "subnet-03f2b6e6ef498e4e9" is already associated with route table "rtb-09d6f14ea76122f8d"
I0105 22:06:44.389917       1 securitygroups.go:35] Reconciling security groups
I0105 22:06:44.547593       1 securitygroups.go:71] Security group for role "bastion": id=sg-075108744364f3201/name=test1-bastion
I0105 22:06:44.547764       1 securitygroups.go:71] Security group for role "controlplane": id=sg-000af8804c0718c1d/name=test1-controlplane
I0105 22:06:44.592295       1 securitygroups.go:71] Security group for role "node": id=sg-099f1b2e964bb698d/name=test1-node
I0105 22:06:44.592490       1 network.go:57] Reconcile network completed successfully
I0105 22:06:44.592669       1 bastion.go:41] Reconciling bastion host
I0105 22:06:44.819576       1 bastion.go:70] Reconcile bastion completed successfully
I0105 22:06:44.819678       1 loadbalancer.go:36] Reconciling load balancers
E0105 22:06:45.995124       1 scope.go:159] [scope] failed to store provider status for cluster "test1" in namespace "default": the server could not find the requested resource (put clusters.cluster.k8s.io test1)
E0105 22:06:45.995229       1 controller.go:131] Error reconciling cluster object test1; unable to reconcile load balancers: failed to create classic load balancer: &{test1-apiserver  Internet-facing [subnet-03f2b6e6ef498e4e9] [sg-000af8804c0718c1d] [0xc0009a6330] 0xc0009a6360 map[kubernetes.io/cluster/test1:owned sigs.k8s.io/cluster-api-provider-aws/managed:true sigs.k8s.io/cluster-api-provider-aws/role:apiserver]}: AccessDenied: User: arn:aws:iam::962279257951:user/bootstrapper.cluster-api-provider-aws.sigs.k8s.io is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::962279257951:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
	status code: 403, request id: 30e830d8-1136-11e9-8530-6936861b7c2a
error: read tcp 192.168.99.1:63084->192.168.99.100:8443: read: operation timed out

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by ashish-amarnath** [comment]
This is the problem
```
map[kubernetes.io/cluster/test1:owned sigs.k8s.io/cluster-api-provider-aws/managed:true sigs.k8s.io/cluster-api-provider-aws/role:apiserver]}: AccessDenied: User: arn:aws:iam::962279257951:user/bootstrapper.cluster-api-provider-aws.sigs.k8s.io is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::962279257951:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
status code: 403, request id: 30e830d8-1136-11e9-8530-6936861b7c2a
```
Can you please delete the cloudformation stack and recreate it using:
```
export AWS_REGION=us-east-1
clusterawsadm alpha bootstrap create-stack
```
All the instructions are available here https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/getting-started.md

**Comment by babu-selector** [comment]
It didn't make much of a difference. Same failure. Please take a look

sigs.k8s.io/cluster-api-provider-aws $ clusterctl create cluster -v2 --provider aws \
>   -m ./cmd/clusterctl/examples/aws/out/machines.yaml \
>   -c ./cmd/clusterctl/examples/aws/out/cluster.yaml \
>   -p ./cmd/clusterctl/examples/aws/out/provider-components.yaml \
>   -a ./cmd/clusterctl/examples/aws/out/addons.yaml
I0106 13:54:05.468915   49273 plugins.go:39] Registered cluster provisioner "aws"
I0106 13:54:05.473386   49273 createbootstrapcluster.go:28] Creating bootstrap cluster
I0106 15:05:31.038553   49273 minikube.go:62] Ran: minikube [start --bootstrapper=kubeadm] Output: There is a newer version of minikube available (v0.32.0).  Download it here:
https://github.com/kubernetes/minikube/releases/tag/v0.32.0

To disable this notification, run the following:
minikube config set WantUpdateNotification false
Starting local Kubernetes v1.10.0 cluster...
Starting VM...
Getting VM IP address...
E0106 13:55:03.702694   49274 start.go:210] Error parsing version semver:  Version string empty
Moving files into cluster...
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Stopping extra container runtimes...
Starting cluster components...
Verifying kubelet health ...
Verifying apiserver health ...Kubectl is now configured to use the cluster.
Loading cached images from config file.


Everything looks great. Please enjoy minikube!
I0106 15:05:31.043976   49273 clusterdeployer.go:95] Applying Cluster API stack to bootstrap cluster
I0106 15:05:31.043989   49273 applyclusterapicomponents.go:27] Applying Cluster API Provider Components
I0106 15:05:31.043999   49273 clusterclient.go:521] Waiting for kubectl apply...
I0106 15:05:31.538771   49273 clusterclient.go:550] Waiting for Cluster v1alpha resources to become available...
I0106 15:05:31.554936   49273 clusterclient.go:563] Waiting for Cluster v1alpha resources to be listable...
I0106 15:05:31.568101   49273 clusterdeployer.go:100] Provisioning target cluster via bootstrap cluster
I0106 15:05:31.576154   49273 applycluster.go:37] Creating cluster object test1 in namespace "default"
I0106 15:05:31.582051   49273 clusterdeployer.go:109] Creating master controlplane-0 in namespace "default"
I0106 15:05:31.588965   49273 applymachines.go:37] Creating machines in namespace "default"
I0106 15:05:31.606769   49273 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...
I0106 15:05:41.609819   49273 clusterclient.go:574] Waiting for Machine controlplane-0 to become ready...





Logs...

I0106 23:28:42.154075       1 controller.go:114] Running reconcile Machine for controlplane-0
I0106 23:28:42.154151       1 actuator.go:236] Checking if machine controlplane-0 for cluster test1 exists
E0106 23:28:42.209812       1 machine_scope.go:129] [machinescope] failed to store provider status for machine "controlplane-0" in namespace "default": the server could not find the requested resource (put machines.cluster.k8s.io controlplane-0)
I0106 23:28:42.209892       1 controller.go:176] Reconciling machine object controlplane-0 triggers idempotent create.
I0106 23:28:42.209925       1 actuator.go:64] Creating machine controlplane-0 for cluster test1
E0106 23:28:42.798228       1 machine_scope.go:129] [machinescope] failed to store provider status for machine "controlplane-0" in namespace "default": the server could not find the requested resource (put machines.cluster.k8s.io controlplane-0)
W0106 23:28:42.798247       1 controller.go:178] unable to create machine controlplane-0: failed to retrieve controlplane url during machine creation: no classic load balancer found with name: "test1-apiserver": LoadBalancerNotFound: There is no ACTIVE Load Balancer named 'test1-apiserver'
	status code: 400, request id: ce12ebbf-120a-11e9-86af-0da450347194
I0106 23:29:17.130336       1 controller.go:89] Running reconcile Cluster for test1
I0106 23:29:17.130360       1 controller.go:124] reconciling cluster object test1 triggers idempotent reconcile.
I0106 23:29:17.130366       1 actuator.go:54] Reconciling cluster test1
I0106 23:29:17.130781       1 network.go:25] Reconciling network
I0106 23:29:17.130789       1 vpc.go:39] Reconciling VPC
I0106 23:29:17.877229       1 vpc.go:54] Working on VPC "vpc-07d9c9b7914bb07cc"
I0106 23:29:17.877251       1 subnets.go:38] Reconciling subnets
I0106 23:29:18.247952       1 subnets.go:116] Subnets available: [id=subnet-01543e17a9d7fa56c/az=us-east-2a/public=false id=subnet-03f2b6e6ef498e4e9/az=us-east-2a/public=true]
I0106 23:29:18.248042       1 gateways.go:33] Reconciling internet gateways
I0106 23:29:18.503715       1 natgateways.go:34] Reconciling NAT gateways
I0106 23:29:18.678415       1 routetables.go:36] Reconciling routing tables
I0106 23:29:18.903498       1 routetables.go:45] Subnet "subnet-01543e17a9d7fa56c" is already associated with route table "rtb-0456c67a8c0fc02e6"
I0106 23:29:18.903517       1 routetables.go:45] Subnet "subnet-03f2b6e6ef498e4e9" is already associated with route table "rtb-09d6f14ea76122f8d"
I0106 23:29:18.903523       1 securitygroups.go:35] Reconciling security groups
I0106 23:29:19.082396       1 securitygroups.go:71] Security group for role "bastion": id=sg-075108744364f3201/name=test1-bastion
I0106 23:29:19.082563       1 securitygroups.go:71] Security group for role "controlplane": id=sg-000af8804c0718c1d/name=test1-controlplane
I0106 23:29:19.082590       1 securitygroups.go:71] Security group for role "node": id=sg-099f1b2e964bb698d/name=test1-node
I0106 23:29:19.082891       1 network.go:57] Reconcile network completed successfully
I0106 23:29:19.083011       1 bastion.go:41] Reconciling bastion host
I0106 23:29:19.403268       1 bastion.go:70] Reconcile bastion completed successfully
I0106 23:29:19.403414       1 loadbalancer.go:36] Reconciling load balancers
E0106 23:29:20.610695       1 scope.go:159] [scope] failed to store provider status for cluster "test1" in namespace "default": the server could not find the requested resource (put clusters.cluster.k8s.io test1)
E0106 23:29:20.610888       1 controller.go:131] Error reconciling cluster object test1; unable to reconcile load balancers: failed to create classic load balancer: &{test1-apiserver  Internet-facing [subnet-03f2b6e6ef498e4e9] [sg-000af8804c0718c1d] [0xc0004cf320] 0xc0004cf350 map[kubernetes.io/cluster/test1:owned sigs.k8s.io/cluster-api-provider-aws/managed:true sigs.k8s.io/cluster-api-provider-aws/role:apiserver]}: AccessDenied: User: arn:aws:iam::962279257951:user/bootstrapper.cluster-api-provider-aws.sigs.k8s.io is not authorized to perform: iam:CreateServiceLinkedRole on resource: arn:aws:iam::962279257951:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing
	status code: 403, request id: e456622a-120a-11e9-9f8e-9ff9239c8597

**Comment by chuckha** [comment]
This is fixed by #422

**Comment by bsingarayan** [comment]
With issue #422 is fixed, it works now. This is not a documentation issue, a real bug
